bridge-fungible-token/src/utils.sw

```diff
@@ -19,14 +28,48 @@ const GTF_INPUT_MESSAGE_DATA = 0x11E;
 const GTF_INPUT_MESSAGE_SENDER = 0x115;
 const GTF_INPUT_MESSAGE_RECIPIENT = 0x116;
 
+fn shift_decimals_left(bn: U256, d: u8) -> Result<U256, BridgeFungibleTokenError> {
+    let mut bn_clone = bn;
+    let mut decimals_to_shift = asm(r1: d) { r1: u64 };
+
+    // the zero case
+    if (decimals_to_shift == 0) {
+        return Result::Ok(bn)
+    };
+
+    // the too large case
+    // (there are only 78 decimal digits in a 256bit number)
+    if (decimals_to_shift > 77) {
+        return Result::Err(BridgeFungibleTokenError::OverflowError)
+    };
+
+    // math time
+    while (decimals_to_shift > 0) {
+        if (decimals_to_shift < 20) {
+            let (prod, overflow) = bn_mult(bn_clone, 10.pow(decimals_to_shift));
+            if (overflow != 0) {
+                return Result::Err(BridgeFungibleTokenError::OverflowError);
+            };
+            return Result::Ok(prod)
```

Comment:
What's the rule in sway for using the `return` keyword? this line doesn't use a semicolon, but the other return lines do

Reply:
good catch. from the book:
>Note return storage.counter; is equivalent to storage.counter.
I will remove the final `return` keyword here as it's not needed.
I will also ask about this in the sway channel as it should maybe be either/or (but not both options) that is supported.